rfcs/0021-ckb-address-format/0021-ckb-address-format.md

```diff
@@ -59,16 +60,35 @@ For example, Alice, Bob, and Cipher collectively control a multisig locked cell.
 0 | 1 | 2 | 3 | Pk_Cipher_h | Pk_Alice_h | Pk_Bob_h
 ```
 
-### Full Payload Format
+Notice that the length of args in payload here is always 20 bytes. So, if you want to append [CKByte minimum field or/and UDT minimum field](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0026-anyone-can-pay/0026-anyone-can-pay.md#script-structure) to anyone_can_pay script, you should use the full payload format. 
 
-Full payload format directly encodes all data field of lock script.
+### Deprecated full Payload Format
+
+Deprecated full payload format directly encodes all data field of lock script.
 
 ```c
 payload = 0x02/0x04 | code_hash | args
 ```
 
 The first byte identifies the lock script's hash_type, 0x02 for "Data", 0x04 for "Type". 
 
+Two reasons have caused this address format to be obsolete. First, a [flaw](https://github.com/sipa/bech32/issues/51) of Bech32 enables attackers to generate valid but unexpected addresses by deleting or inserting characters into certain full addresses. Last, a new field is needed to indicate the CKB VM version for each script group.
+
+### New full Payload Format 
+
+New full payload format add two fields: `hash_type` and `args_len`. 
+
+```c
+payload = 0x08 | code_hash | hash_type | args_len (fixed 2 bytes) | args
+```
+
+The `hash_type` field is for CKB VM version selection. 
```

Comment:
We have updated the vm version selection sepc in rfc 0032, please update accordingly:
https://github.com/nervosnetwork/rfcs/blob/dd5f26a02d3db1659f86c42bdbf52c49975a68bb/rfcs/0032-ckb-vm-version-selection/0032-ckb-vm-version-selection.md#specification